File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.go
func maxAscendingSum(nums []int) int {
    
    currSum := nums[0]

    maxSum := nums[0]
    
    for i:=1; i < len(nums); i++ {

        if nums[i-1] < nums[i] {
            currSum += nums[i]
        } else {
            currSum = nums[i]
        }

        if maxSum < currSum {
            maxSum = currSum
        }

    }


    return maxSum;
}

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        
        int max = nums[0], result = 0;
        boolean isFirst = true;

        for (int i = 1; i < nums.length; i++ ) {
            
            if (isFirst) {
                max = nums[i-1];
            }

            if (nums[i-1] < nums[i]) {
                max += nums[i];
                isFirst = false;
            } else {
                if (!isFirst) {
                    max = 0;
                    isFirst = true;
                }
            }

            result = Math.max(result, max);

        }

        return Math.max(result, max);

    }

}